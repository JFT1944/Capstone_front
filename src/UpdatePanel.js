import React, {useEffect, useState} from "react";
import { useNavigate, useParams } from "react-router-dom";
import DispensaApi from "./ApiHelper";


function UpdatePanel(props){
const navigate = useNavigate()
let {user, token} = props
let ingredientForm = useParams()
console.log(ingredientForm)
let [ing, getIng] = useState(undefined)
let [name, getName] = useState('ingredient')

useEffect(() =>{
console.log(ingredientForm)
try {
    DispensaApi.getIngredient(user.username, ingredientForm.ingredientForm).then((res) => {
        console.log(res)
        getIng(res)
        getName(res.name)
    })
} catch (error) {
    console.log(error)
}

}, [])




const [formData, getFormData] = useState({
    available_amount: '', 
    full_amount: ''
})
// let {tokenState, token, user, userState} = props
// console.log(token)

const handleChange = (evt) =>{

    // updatedFormData(evt.target.value)
    const {name, value} = evt.target;
    console.log(evt)
    getFormData(data => ({
        ...data,
        [name]: value
    }))
}


function handleClick(e){
e.preventDefault()
console.log('clicked')
console.log({available_amount:formData.available_amount, full_amount:formData.full_amount})
console.log({id:ing.id})
DispensaApi.updateIngredient(ing.id, formData).then((res) => {
    console.log(res)
    navigate('/pantry')
})
}

if(!ing){
    navigate('/pantry')
}

    return(
        <>
       <div className="updatePanelHolder">
       {/* updatePanelHolder */}
            <div className="">
                <form className="iForms">
                    <h1>Update Your Ingredient</h1>
                    <span><input type={'number'} name={'available_amount'}  value={formData.available_amount} placeholder={`Newly Available Amount for ${name} (Number Only)`} onChange={(e) => handleChange(e)} /></span>
                    <span><input type={'number'} name={'full_amount'} value={formData.full_amount} placeholder={`New Total Amount for ${name} (Number Only)`} onChange={(e) => handleChange(e)} /></span>
                    <span className="submit_button" onClick={(e) => handleClick(e)}><input type={'submit'} /></span>
                </form>
            </div>
       </div>
        </>
    )
} 



export default UpdatePanel