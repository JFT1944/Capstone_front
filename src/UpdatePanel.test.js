
import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import UpdatePanel from "./UpdatePanel";
import { BrowserRouter, MemoryRouter } from "react-router-dom";


// write a smoke test for the home component

it('should appear in a smoketest', () => {
    <BrowserRouter>
    render(<UpdatePanel user={'user'} setUser={'setUser'} token={'token'} setToken={'setToken'}/>)
    </BrowserRouter>
})

it('should match snapshot', () => {
    const {asFragment} = render(<MemoryRouter><UpdatePanel user={'user'} setUser={'setUser'} token={'token'} setToken={'setToken'}/></MemoryRouter>)
    expect(asFragment()).toMatchSnapshot()
})