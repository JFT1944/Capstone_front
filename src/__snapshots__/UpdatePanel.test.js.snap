// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="updatePanelHolder"
  >
    <div
      class=""
    >
      <form
        class="iForms"
      >
        <h1>
          Update Your Ingredient
        </h1>
        <span>
          <input
            name="available_amount"
            placeholder="Newly Available Amount for ingredient (Number Only)"
            type="number"
            value=""
          />
        </span>
        <span>
          <input
            name="full_amount"
            placeholder="New Total Amount for ingredient (Number Only)"
            type="number"
            value=""
          />
        </span>
        <span
          class="submit_button"
        >
          <input
            type="submit"
          />
        </span>
        <span
          class="submit_button"
        >
          <button>
            Delete Ingredient
          </button>
        </span>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
